src/Common/src/CoreLib/System/IO/Stream.cs

```diff
@@ -785,15 +781,11 @@ public virtual void Write(ReadOnlySpan<byte> buffer)
         }
 
         // Writes one byte from the stream by calling Write(byte[], int, int).
-        // This implementation does not perform well because it allocates a new
-        // byte[] each time you call it, and should be overridden by any 
-        // subclass that maintains an internal buffer.  Then, it can help perf
-        // significantly for people who are writing one byte at a time.
         public virtual void WriteByte(byte value)
         {
-            byte[] oneByteArray = new byte[1];
-            oneByteArray[0] = value;
-            Write(oneByteArray, 0, 1);
+            Span<byte> oneByteSpan = stackalloc byte[1];
+            oneByteSpan[0] = value;
+            Write(oneByteSpan);
```

Comment:
Ditto.

Reply:
Reverted in https://github.com/dotnet/corefx/pull/36838/commits/1daf838d3f5a474856e9c3503604ba37758ad8fc